front/app/modules/commercial/content_builder/admin/components/CraftComponents/Iframe/index.tsx

```diff
@@ -0,0 +1,197 @@
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeUrlLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeUrlLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeUrlPlaceholder)}
+          type="text"
+          value={url}
+          onChange={(value) => {
+            const validation = validateUrl(value);
+            setProp((props) => (props.url = value));
+            setProp((props) => (props.errorType = validation[1]));
+            setProp((props) => (props.hasError = !validation[0]));
+            eventEmitter.emit('contentBuilderError', {
+              [id]: !validation[0],
+            });
+          }}
+        />
+        {hasError && (
+          <Error
+            marginTop="4px"
+            text={
+              errorType === 'invalidUrl'
+                ? formatMessage(messages.iframeInvalidUrlErrorMessage)
+                : formatMessage(messages.iframeWhitelistUrlErrorMessage)
+            }
+          />
+        )}
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeHeightLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeHeightLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeHeightPlaceholder)}
+          type="text"
+          value={height}
+          onChange={(value) => {
+            setProp((props) => (props.height = value));
+          }}
+        />
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          type="text"
+          id="iframeTitleInput"
+          onChange={(value) => {
+            setProp((props) => (props.title = value));
+          }}
+          value={title}
+          label={
+            <span>
+              {formatMessage(messages.iframeTitleLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeTitleTooltip)}
+              />
+            </span>
+          }
+        />
+      </Box>
+    </Box>
+  );
+});
+
+Iframe.craft = {
+  props: {
+    url: '',
+    height: '',
+  },
+  related: {
+    settings: IframeSettings,
+  },
+};
+
+/*
+ * Function to validate embed URL against white list
+ * Returns: boolean value whether URL input string is valid or not
+ */
+const validateUrl = (url: string) => {
+  // Used this reference for generating a valid URL regex:
+  // https://tutorial.eyehunts.com/js/url-regex-validation-javascript-example-code/
+  const validUrlCheck = url.match(
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g
+  );
+  if (validUrlCheck === null) {
```

Comment:
```suggestion
  if (!validUrlCheck) {
```

Reply:
@koendehondt I believe string.match is the correct function here (matches a string against a regular expression)? In which case it would be null rather than a boolean returned for something that doesn't pass. Or can I use an exclamation mark for this kind of check as well?

Reply:
The return possible return types of `string.match` are `RegExpMatchArray | null`, so since `RegExpMatchArray` is truthy you can get away with just doing `!validUrlCheck`. It's a bit more concise, but not a huge blocker :)

Reply:
@amanda-anderson I agree with Luuc that it is not a blocker. That is not why I made the comment. In principle a linter should indicate that there is useless code, but probably it didn't. Useless code makes code harder to read. What you like to express is "if the URL is not valid then ...". If you refactor the code a bit, then it reads just like that:

```
const invalidUrl = !url.match(
    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g
  );
if (invalidUrl) {
  ...
}
```
In general, try to write code as if you would in plain English. It will help choosing good names for variables, which results in better abstractions, and code that is easier to read. Code is more read than written, so take time to express things as elegantly as possible.